File: api/server/app/assets/modelValidator/server/src/main/java/modelValidator/ValidationResult.java
package assets.modelValidator.server.src.main.java.modelValidator;


import java.util.List;

public class ValidationResult {

    private final boolean valid;
    private final List<String> messages;

    public ValidationResult (boolean valid, List<String> messages) {
        this.valid = valid;
        this.messages = messages;
    }

    public boolean isValid () {
        return valid;
    }

    public List<String> getMessages () {
        return messages;
    }

}

File: webapp/src/yfiles-editor/export/ValidationResult.js
export default (function() {
  function ValidationResult() {
    this.valid = true;
    this.messages = [];
  }

  ValidationResult.prototype.addErrorMessage = function(message) {
    this.valid = false;
    return this.messages.push(message);
  };

  ValidationResult.prototype.isValid = function() {
    return this.valid;
  };

  ValidationResult.prototype.getMessages = function() {
    return this.messages;
  };

  return ValidationResult;
})();
